src/api/AutonomousCars.Api/Itinerary/Services/AzureMapsItineraryService.cs

```diff
@@ -63,33 +65,40 @@ public Feature<LineString> ComputeStatusData(string carId)
         };
         return new Feature<LineString>(positionsLineString, properties);
     }
-    
     public async Task SendRequest(CancellationToken stoppingToken,  Feature<LineString> geospatialFeature, bool isStatusRequest)
     {
-        var cs = MqttConnectionSettings.CreateFromEnvVars(_configuration.GetValue<string>("envFile"));
-        _logger.LogInformation("Connecting to {cs}", cs);
-
-        var mqttClient = new MqttFactory().CreateMqttClient(MqttNetTraceLogger.CreateTraceLogger());
-        mqttClient.DisconnectedAsync += e =>
+        
+        var cs = MqttSettings.MqttConnectionSettings;
+        if (cs != null)
```

Comment:
Toute petite remarque, en général c'est plus lisible de tester le cas de sortie immédiate/erreur, ca retire des imbrications de if else 

Par exemple ici tu pourrais faire au tout début 
if (cs is null)
{
   _logger.LogError("Error to open the connexion");
   return;
}

// Le reste de ton code, sans else